reference/regexp_test.go

```diff
@@ -0,0 +1,171 @@
+package reference
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func TestHostRegexp(t *testing.T) {
+	hostcases := map[string]bool{
```

Comment:
nit: leave these as struct style test cases. They will be easier to convert if we add more testing later.